service/history/service.go

```diff
@@ -90,6 +100,11 @@ func (s *Service) Start() {
 	s.handler.Start()
 
 	historyservice.RegisterHistoryServiceServer(s.server, s.handler)
+
+	for _, hook := range s.handlerRegistrationHooks {
+		hook(s.server)
```

Comment:
this seems a little ad-hoc.. is there a way to do it by exposing the grpc server in the fx graph, e.g. use grpc.NewServer as a provider, and then something else added to the fx container can grab the *grpc.Server and register itself? you just have to figure out the lifecycle bit to order it properly

Reply:
Ack. Let me try that.

Reply:
I was able to provide a `grpc.Server` with a handler registered to it using `fx.Invoke()`. For my use case, all I need here is an fx provider for `grpc.Server`. This way it is much simpler as well. Thanks for the suggestion!

Reply:
Is the ordering a problem? According to docs it looks like you have to register all services before Serve() is called.

Oh, I suppose all Invoke calls will happen before Start, since Start is from a lifecycle hook. So yeah, that should be fine.